### PR TITLE
Refactor compare image

### PR DESCRIPTION
**Pull request by danrr**

### CODE REVIEW

File: src/compareImage.js [hunk 1]
```diff
@@ -1,33 +1,48 @@
...
+  try {
+    testImage = await Jimp.read(testFile);
+  } catch (err) {
+    throw new Error(`${testName}: failed to read test image ${err}`);
+  }
+
+  const distance = Jimp.distance(referenceImage, testImage);
+  const diff = Jimp.diff(referenceImage, testImage, options.mismatchThreshold);
+  if (distance < options.mismatchThreshold || diff.percent < options.mismatchThreshold) {
+    return `${testName}: no mismatch found ✅`;
```
Comment (NimaSoroush) [inline src/compareImage.js]: I like this ✅

File: .gitignore [hunk 2]
```diff
@@ -32,6 +32,5 @@ dist/
 # Built docs
 docs/
 .vscode/
-dist/
```
Comment (danrr) [inline .gitignore]: duplicate

File: src/chromyRunner.test.js [hunk 3]
```diff
@@ -69,7 +69,6 @@ describe('ChromyRunner', () => {
       expect(loggerCalls[0]).toEqual('goto -> www.example.com');
       expect(loggerCalls[1]).toEqual('Capturing screenshot of whole DOM');
       expect(loggerCalls[2]).toEqual('screenshot saved in -> ./differencify_report/default.png');
-      expect(loggerCalls[4]).toEqual('Saving the diff image to disk');
```
Comment (danrr) [inline src/chromyRunner.test.js]: testing logs is not a good way of verifying behaviour, so I didn't update it. `src/chromyRunner.test.js` needs a big refactor IMO

File: src/index.test.js [hunk 4]
```diff
@@ -78,7 +78,6 @@ describe('Differencify', () => {
     expect(loggerCalls[0]).toEqual('goto -> www.example.com');
     expect(loggerCalls[1]).toEqual('Capturing screenshot of whole DOM');
     expect(loggerCalls[2]).toEqual('screenshot saved in -> ./differencify_report/default.png');
-    expect(loggerCalls[4]).toEqual('Saving the diff image to disk');
```
Comment (danrr) [inline src/index.test.js]: ditto

File: src/chromyRunner.test.js (hunk 3 above)
Comment (NimaSoroush) [inline src/chromyRunner.test.js]: Totally agree that relying on verifying logs is not enough to validate your test, but it is a good safe guard beside other functionalities IMO

**Pull request merged**